src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +240,114 @@ pub fn register(registry: &mut FunctionRegistry) {
         |map, _| map.len() as u64,
     );
 
+    registry.register_function_factory("map_delete", |_, args_type| {
+        if args_type.len() < 2 {
+            return None;
+        }
+
+        let map_key_type = match args_type[0].remove_nullable() {
+            DataType::Map(box DataType::Tuple(type_tuple)) if type_tuple.len() == 2 => {
```

Comment:
We need support `EmptyMap` here, the previous code is correct, you can modify it back. Delete `DataType::EmptyMap` on line 317 is because we have already processed the `EmptyMap` type on line 299, so `EmptyMap` type can't appear on line 317. 
I think there is basically ok with the rest of the codes. After this modification is completed, we can merge this PR.

Reply:
Thankyou

Reply:
Hi @shamb0 , seems not correct. `select map_delete({}, 'a')` will return `SemanticError`, you can write like this:
```rust
let map_key_type = match args_type[0].remove_nullable() {
    DataType::Map(box DataType::Tuple(type_tuple)) if type_tuple.len() == 2 => {
        Some(type_tuple[0].clone())
    }
    DataType::EmptyMap => None,
    _ => return None,
};

if let Some(map_key_type) = map_key_type {
    for arg_type in args_type.iter().skip(1) {
        if arg_type != &map_key_type {
            return None;
        }
    }
} else {
    let key_type = &args_type[1];
    if !key_type.is_boolean()
        && !key_type.is_string()
        && !key_type.is_numeric()
        && !key_type.is_decimal()
        && !key_type.is_date_or_date_time()
    {
        return None;
    }
    for arg_type in args_type.iter().skip(2) {
        if arg_type != key_type {
            return None;
        }
    }
}
```
Please also remove `DataType::EmptyMap => DataType::EmptyMap` at line 325, because this case will never matched.

Add a unit test for `EmptyMap`, like `map_delete({}, 'a', 'b')`. After these changes are complete, we can merge this PR.

Reply:
Hi @b41sh,

Changes are done, and I've added the suggested test case. Could you please double-check the result? It looks correct on my end.

```txt
ast            : map_delete({}, 'a', 'b')
raw expr       : map_delete(map(array(), array()), 'a', 'b')
checked expr   : map_delete<Map(Nothing), String, String>(map<Array(Nothing), Array(Nothing)>(array<>(), array<>()), "a", "b")
optimized expr : {} :: Map(Nothing)
output type    : Map(Nothing)
output domain  : {}
output         : {}
```